src/ignore.c

```diff
@@ -41,9 +76,12 @@ static int does_negate_rule(int *out, git_vector *rules, git_attr_fnmatch *match
 	path = git_buf_detach(&buf);
 
 	git_vector_foreach(rules, i, rule) {
-		/* no chance of matching w/o a wilcard */
-		if (!(rule->flags & GIT_ATTR_FNMATCH_HASWILD))
-			continue;
+		if (!(rule->flags & GIT_ATTR_FNMATCH_HASWILD)) {
+			if (does_negate_pattern(rule, match))
+				goto out;
```

Comment:
One pattern that we try to follow is to not make callers necessarily initialize the out param -- it would be nice if we moved the `*out = 0` to the first instruction in the func in keeping with this pattern, or assign it here.  Now it appears to assume that it was set appropriately by the caller...